src/lpython/semantics/python_ast_to_asr.cpp

```diff
@@ -4356,6 +4356,65 @@ class BodyVisitor : public CommonVisitor<BodyVisitor> {
                                     1, 1, nullptr, nullptr , 0));
                     tmp = ASR::make_StringConstant_t(al, x.base.base.loc, s2c(al, res), str_type);
                     return;
+                } else if (std::string(at->m_attr) == std::string("find")) {
+                    if (args.size() != 1) {
+                        throw SemanticError("str.find() takes one arguments",
+                            x.base.base.loc);
+                    }
+                    ASR::expr_t *arg = args[0].m_value;
+                    ASR::ttype_t *type = ASRUtils::expr_type(arg);
+                    if (ASRUtils::is_character(*type)) {
+                        std::string str = (ASR::down_cast<ASR::StringConstant_t>(at->m_value))->m_s;
+                        if (ASRUtils::expr_value(arg) != nullptr) {
+                            ASR::StringConstant_t* ss = ASR::down_cast<ASR::StringConstant_t>(ASRUtils::expr_value(arg));
```

Comment:
@certik @czgdp1807 line i mentioned on zulip

Reply:
Above at line number `4367`: `at->value` is of type `LFortran::LPython::AST:expr_t *`. I think we have downcast it using `AST::down_cast<AST::ConstantStr_t>(at->m_value)`, just like we did in line `4348`

Reply:
@Thirumalai-Shaktivel 
i think casting works with AST and ASR, i tried them and line 4369 still raise the same error
```bash
In file included from /home/abdelrahmankh/Gsoc/lpython/src/libasr/alloc.h:10,
                 from /home/abdelrahmankh/Gsoc/lpython/src/libasr/asr.h:6,
                 from /home/abdelrahmankh/Gsoc/lpython/src/lpython/semantics/python_ast_to_asr.cpp:13:
/home/abdelrahmankh/Gsoc/lpython/src/lpython/python_ast.h: In instantiation of ‘T* LFortran::LPython::AST::down_cast(const U*) [with T = LFortran::ASR::StringConstant_t; U = LFortran::ASR::expr_t]’:
/home/abdelrahmankh/Gsoc/lpython/src/lpython/semantics/python_ast_to_asr.cpp:4369:94:   required from here
/home/abdelrahmankh/Gsoc/lpython/src/lpython/python_ast.h:40:28: error: call of overloaded ‘is_a<LFortran::ASR::StringConstant_t>(const LFortran::ASR::expr_t&)’ is ambiguous
   40 |     LFORTRAN_ASSERT(is_a<T>(*f));

```

Reply:
yes, it works now, thanks @Thirumalai-Shaktivel .